### DISCUSSION TITLE
carrier frequency, me confused

### DISCUSSION THREAD

**Discussion by KingJimmy27** [bengtmartensson/Infrared4Arduino#61] [General]
I am developping an extended IR controller for a NAD amplifier which uses NEC1 protocoll which I understand uses an IR carrier frequency of 38kHz.

The controller supplied with the amplifier uses an AD009-02T proccessor to generate the code. 
I have measured the signal using a TSMP58000 IR sensor (with 4.7k pullup attached) using a PC digital oscilloscope. 
The measured carrier frequency is 38.2 kHz. 
Scrutinizer reports NEC1 codes with 38461 kHz.

I have generated an Arduino .ino file using Scrutinizer running on an Arduino Nano.
The oscilloscope reports 38.51 kHz.
Scrutinizer reports NEC1 codes with 38461 kHz.

I have also tested the SimpleSender.ino example from [https://github.com/ukw100/IRMP](url) 
in which:
#define IRSND_IR_FREQUENCY          38000
The oscilloscope reports 38.23 kHz.
Scrutinizer reports NEC1 codes with 38461 kHz.

I notice in the Infrared4Arduino example "IrSenderPwm.ino" that:
static const frequency_t necFrequency = 38400U;

**Question**, why is the NEC1 carrier set to 38400 and not 38000 in Infrared4Arduino - I accept that the measured frequencies will vary a bit from the set frequency.
I have tried to follow the Infrared4Arduino code to see where the carrier is set but my programming skills are not up to it.

**Comment by bengtmartensson** [comment]
To my knowledge, 38400 comes from the original specification from the inventor of the procol, the company NEC (now Renesas). So I consider this the correct value. Anyhow, the difference is around 1%, and practically not of relevance: see for example Fig 5. in the [data sheet of a typical receiver](https://www.vishay.com/docs/82489/tsop322.pdf) (page 3).

For Infrared4Arduino's Nec1-implementation: The declaration is in `src/Nec1Renderer.h`:

    static const frequency_t frequency = 38400U;
